storage/src/main/java/org/apache/kafka/storage/internals/log/LogAppendInfo.java

```diff
@@ -204,6 +204,10 @@ public LeaderHwChange leaderHwChange() {
         return leaderHwChange;
     }
 
+    public boolean hasValidBytes() {
```

Comment:
Could this be called `hasMessages` and be used for both cases in this PR?

Reply:
one reason I can think of not to change the `KafkaMetadataLog` usage is because the `raft` package `LogAppendInfo` relies on firstOffset and lastOffset being set, which the current `numMessages` call validates.

On the other hand, firstOffset and lastOffset are not set by the time we need the validBytes check.

I am OK with renaming this to `hasMessages`. Let me know what you think.